test/distributed/_tensor/test_dtensor.py

```diff
@@ -230,6 +230,23 @@ def test_to_local(self):
 
         self.assertEqual(sharded_tensor.grad.to_local(), torch.ones(3, 3) * 3)
 
+        # test the case when grad stride is different from fwd input.
+        res = sharded_tensor.to_local()
+        model = torch.nn.ReLU()
+        res.register_hook(lambda grad: grad.t())
+        sharded_tensor.register_hook(
```

Comment:
why this needs a hook registration, instead of just run the backward(), then compare `sharded_tensor.grad.stride(), [1, 12]`?

Reply:
Originally the check failed in the end... Looks like it's working now. Let me remove it.